channels/tsmf/client/tsmf_main.c

```diff
@@ -104,6 +95,15 @@ BOOL tsmf_playback_ack(IWTSVirtualChannelCallback *pChannelCallback,
 				Stream_GetPosition(s), Stream_Buffer(s), NULL);
 	}
 
+        Stream_Write_UINT32(s, TSMF_INTERFACE_CLIENT_NOTIFICATIONS | STREAM_ID_PROXY);
+        Stream_Write_UINT32(s, message_id);
+        Stream_Write_UINT32(s, PLAYBACK_ACK); /* FunctionId */
+        Stream_Write_UINT32(s, callback->stream_id); /* StreamId */
```

Comment:
Still using unchecked access to callback here. Better revert and just do an additional check ```if (!callback) return FALSE;````before the stream block.